models/mart/core/fct_orders.sql

```diff
@@ -0,0 +1,11 @@
+
+with orders as (
+    SELECT * FROM {{ref('stg_jaffle_shop__orders')}}
+)
+
+SELECT o.*
+    ,sum(p.amount) as amount
+FROM orders o
+LEFT JOIN {{ref('stg_stripe_payments')}} p
+ON o.order_id = p.order_id
+GROUP BY 1,2,3,4
```

Comment:
Another small style thing: `group by 1,2,3,4` will break if the number of columns in `orders` changes (as pulled through by `o.*`). My inclination for aggregation CTEs is to always list out those columns explicitly.

Reply:
What I'll often do to avoid this altogether is have a separate CTE that aggregates first and then use that aggregated CTE in the join, e.g.
```sql
total_paid as (

    select

        order_id,
        sum(amount) as amount

    from payments
    group by 1

),

joined as (

    select * from orders
    left join total_paid using (order_id)

)

select * from joined
```